Jellyfin.Api/Controllers/MediaSegmentController.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using Jellyfin.Api.Extensions;
+using MediaBrowser.Controller;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Library;
+using MediaBrowser.Model.MediaSegments;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Jellyfin.Api.Controllers;
+
+/// <summary>
+/// Media Segments api.
+/// </summary>
+[Authorize]
+public class MediaSegmentController : BaseJellyfinApiController
+{
+    private readonly IMediaSegmentManager _mediaSegmentManager;
+    private readonly ILibraryManager _libraryManager;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MediaSegmentController"/> class.
+    /// </summary>
+    /// <param name="mediaSegmentManager">MediaSegments Manager.</param>
+    /// <param name="libraryManager">The Library manager.</param>
+    public MediaSegmentController(IMediaSegmentManager mediaSegmentManager, ILibraryManager libraryManager)
+    {
+        _mediaSegmentManager = mediaSegmentManager;
+        _libraryManager = libraryManager;
+    }
+
+    /// <summary>
+    /// Gets all media segments based on an itemId.
+    /// </summary>
+    /// <param name="itemId">The ItemId.</param>
+    /// <returns>A list of media segement objects related to the requested itemId.</returns>
+    [HttpGet("MediaSegments/{itemId}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public ActionResult<IAsyncEnumerable<MediaSegmentModel>> GetSegmentsAsync([FromRoute, Required] Guid itemId)
```

Comment:
Didn't think of this in the meta issue but it would be good to add an (optional) filter `includeSegmentTypes` that, if specified, filters to only the requested segment types. Otherwise return all.

Reply:
hmm what would be the usecase for that?

Reply:
- A client may not support all segment types, for example by not having an implementation (e.g. the annotation type from the possible extensions)
- A client may want to list the segment for an item and have a filter for the user to only show specific types